<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 smokingMouse RSE | Statistical Analysis of Genome Scale Data 2023</title>
  <meta name="description" content="4 smokingMouse RSE | Statistical Analysis of Genome Scale Data 2023" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="4 smokingMouse RSE | Statistical Analysis of Genome Scale Data 2023" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lcolladotor/cshl_rstats_genome_scale_2023" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 smokingMouse RSE | Statistical Analysis of Genome Scale Data 2023" />
  
  
  

<meta name="author" content="Leonardo Collado-Torres" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />
<link rel="prev" href="summarizedexperiment-review.html"/>
<link rel="next" href="isee-introduction.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#code-of-conduct"><i class="fa fa-check"></i>Code of Conduct</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-schedule"><i class="fa fa-check"></i>Course Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#external-links"><i class="fa fa-check"></i>External links</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-prerequisites"><i class="fa fa-check"></i>Course Prerequisites</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html"><i class="fa fa-check"></i><b>1</b> SPEAQeasy introduction</a></li>
<li class="chapter" data-level="2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html"><i class="fa fa-check"></i><b>2</b> smokingMouse study</a>
<ul>
<li class="chapter" data-level="2.1" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#overview-1"><i class="fa fa-check"></i><b>2.2</b> Overview</a></li>
<li class="chapter" data-level="2.3" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#goals"><i class="fa fa-check"></i><b>2.3</b> Goals</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#main-objective"><i class="fa fa-check"></i><b>2.3.1</b> Main objective</a></li>
<li class="chapter" data-level="2.3.2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#secondary-objectives"><i class="fa fa-check"></i><b>2.3.2</b> Secondary objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#study-design"><i class="fa fa-check"></i><b>2.4</b> Study design</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html"><i class="fa fa-check"></i><b>3</b> SummarizedExperiment review</a>
<ul>
<li class="chapter" data-level="3.1" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#overview-2"><i class="fa fa-check"></i><b>3.1</b> Overview</a></li>
<li class="chapter" data-level="3.2" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#quiz"><i class="fa fa-check"></i><b>3.2</b> Quiz</a></li>
<li class="chapter" data-level="3.3" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#exercises"><i class="fa fa-check"></i><b>3.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html"><i class="fa fa-check"></i><b>4</b> smokingMouse RSE</a>
<ul>
<li class="chapter" data-level="4.1" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#download-data"><i class="fa fa-check"></i><b>4.1</b> Download data</a></li>
<li class="chapter" data-level="4.2" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#data-overview"><i class="fa fa-check"></i><b>4.2</b> Data overview</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#assays"><i class="fa fa-check"></i><b>4.2.1</b> Assays</a></li>
<li class="chapter" data-level="4.2.2" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#sample-data"><i class="fa fa-check"></i><b>4.2.2</b> Sample data</a></li>
<li class="chapter" data-level="4.2.3" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#gene-information"><i class="fa fa-check"></i><b>4.2.3</b> Gene Information</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="isee-introduction.html"><a href="isee-introduction.html"><i class="fa fa-check"></i><b>5</b> iSEE introduction</a></li>
<li class="chapter" data-level="6" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html"><i class="fa fa-check"></i><b>6</b> RSE plotting basics with smokingMouse</a>
<ul>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#data-preparation"><i class="fa fa-check"></i>1. Data preparation</a>
<ul>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#data-normalization"><i class="fa fa-check"></i>1.1 Data normalization</a></li>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#gene-filtering"><i class="fa fa-check"></i>1.2 Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#exploratory-data-analysis"><i class="fa fa-check"></i>2. Exploratory Data Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#quality-control-analysis"><i class="fa fa-check"></i>2.1 Quality Control Analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#references"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html"><i class="fa fa-check"></i><b>7</b> biocthis introduction</a></li>
<li class="chapter" data-level="8" data-path="recount3-introduction.html"><a href="recount3-introduction.html"><i class="fa fa-check"></i><b>8</b> recount3 introduction</a></li>
<li class="chapter" data-level="9" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html"><i class="fa fa-check"></i><b>9</b> Interpreting model coefficients with ExploreModelMatrix</a></li>
<li class="chapter" data-level="10" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html"><i class="fa fa-check"></i><b>10</b> Differential gene expression analysis with limma</a></li>
<li class="chapter" data-level="11" data-path="differential-gene-expression-exercise-with-recount3-data.html"><a href="differential-gene-expression-exercise-with-recount3-data.html"><i class="fa fa-check"></i><b>11</b> Differential gene expression exercise with recount3 data</a></li>
<li class="chapter" data-level="12" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html"><i class="fa fa-check"></i><b>12</b> DE model building with variancePartition</a>
<ul>
<li class="chapter" data-level="" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html#explore-gene-level-effects"><i class="fa fa-check"></i>Explore gene-level effects</a>
<ul>
<li class="chapter" data-level="" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html#variance-partition-analysis"><i class="fa fa-check"></i>1. Variance Partition Analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html#references-1"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html"><i class="fa fa-check"></i><b>13</b> Expression heatmaps with ComplexHeatmap</a></li>
<li class="chapter" data-level="14" data-path="exercise-with-variancepartition-and-complexheatmap.html"><a href="exercise-with-variancepartition-and-complexheatmap.html"><i class="fa fa-check"></i><b>14</b> Exercise with variancePartition and ComplexHeatmap</a></li>
<li class="chapter" data-level="15" data-path="tbd.html"><a href="tbd.html"><i class="fa fa-check"></i><b>15</b> TBD</a></li>
<li class="chapter" data-level="16" data-path="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><a href="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><i class="fa fa-check"></i><b>16</b> Introduction to the world of spatially-resolved transcriptomics with Visium data</a></li>
<li class="chapter" data-level="17" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><i class="fa fa-check"></i><b>17</b> Downloading public Visium spatial data and visualizing it with spatialLIBD</a></li>
<li class="chapter" data-level="18" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><i class="fa fa-check"></i><b>18</b> Re-use of bulk RNA-seq methods for spatial data exercise</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Genome Scale Data 2023</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="smokingmouse-rse" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">4</span> smokingMouse RSE<a href="smokingmouse-rse.html#smokingmouse-rse" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Instructor: Daianna</p>
<p>Once you have reviewed how to use <code>RangedSummarizedExperiment</code> (RSE) objects in R, now we’ll start exploring the <code>smokingMouse</code> data.</p>
<p>As previously described, the smoking mouse project is a complex study with more than 200 samples from brain and blood, adult mice and pups, pregnant and not pregnant mice, etc. The whole datasets of this project can be downloaded from the <code>smokingMouse</code> (Gonzalez-Padilla and Collado-Torres, 2023) package. Visit <a href="http://research.libd.org/smokingMouse/">here</a> for more details.</p>
<div id="download-data" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Download data<a href="smokingmouse-rse.html#download-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For illustrative purposes, we’ll use nicotine data at the gene level which reside in a <code>RangedSummarizedExperiment</code> (RSE) object called <code>rse_gene</code>. We’ll use <em><a href="https://bioconductor.org/packages/3.17/BiocFileCache">BiocFileCache</a></em> to keep the data in a local cache in case we want to run this example again and don’t want to re-download the data from the web.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="smokingmouse-rse.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Download and cache the file</span></span>
<span id="cb12-2"><a href="smokingmouse-rse.html#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;BiocFileCache&quot;</span>)</span>
<span id="cb12-3"><a href="smokingmouse-rse.html#cb12-3" aria-hidden="true" tabindex="-1"></a>bfc <span class="ot">&lt;-</span> BiocFileCache<span class="sc">::</span><span class="fu">BiocFileCache</span>()</span>
<span id="cb12-4"><a href="smokingmouse-rse.html#cb12-4" aria-hidden="true" tabindex="-1"></a>cached_rse_gene <span class="ot">&lt;-</span> BiocFileCache<span class="sc">::</span><span class="fu">bfcrpath</span>(</span>
<span id="cb12-5"><a href="smokingmouse-rse.html#cb12-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> bfc,</span>
<span id="cb12-6"><a href="smokingmouse-rse.html#cb12-6" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;https://github.com/LieberInstitute/SPEAQeasyWorkshop2023/raw/devel/provisional_data/rse_gene_mouse_RNAseq_nic-smo.Rdata&quot;</span></span>
<span id="cb12-7"><a href="smokingmouse-rse.html#cb12-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb12-8"><a href="smokingmouse-rse.html#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; adding rname &#39;https://github.com/LieberInstitute/SPEAQeasyWorkshop2023/raw/devel/provisional_data/rse_gene_mouse_RNAseq_nic-smo.Rdata&#39;</span></span>
<span id="cb12-9"><a href="smokingmouse-rse.html#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="smokingmouse-rse.html#cb12-10" aria-hidden="true" tabindex="-1"></a><span class="do">## Check the local path on our cache</span></span>
<span id="cb12-11"><a href="smokingmouse-rse.html#cb12-11" aria-hidden="true" tabindex="-1"></a>cached_rse_gene</span>
<span id="cb12-12"><a href="smokingmouse-rse.html#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                                                                                  BFC1 </span></span>
<span id="cb12-13"><a href="smokingmouse-rse.html#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; &quot;/github/home/.cache/R/BiocFileCache/55c3c13e47e_rse_gene_mouse_RNAseq_nic-smo.Rdata&quot;</span></span>
<span id="cb12-14"><a href="smokingmouse-rse.html#cb12-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-15"><a href="smokingmouse-rse.html#cb12-15" aria-hidden="true" tabindex="-1"></a><span class="do">## Load the rse_gene object</span></span>
<span id="cb12-16"><a href="smokingmouse-rse.html#cb12-16" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(cached_rse_gene, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-17"><a href="smokingmouse-rse.html#cb12-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Loading objects:</span></span>
<span id="cb12-18"><a href="smokingmouse-rse.html#cb12-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   rse_gene</span></span>
<span id="cb12-19"><a href="smokingmouse-rse.html#cb12-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-20"><a href="smokingmouse-rse.html#cb12-20" aria-hidden="true" tabindex="-1"></a><span class="do">## General overview of the object</span></span>
<span id="cb12-21"><a href="smokingmouse-rse.html#cb12-21" aria-hidden="true" tabindex="-1"></a>rse_gene</span>
<span id="cb12-22"><a href="smokingmouse-rse.html#cb12-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; class: RangedSummarizedExperiment </span></span>
<span id="cb12-23"><a href="smokingmouse-rse.html#cb12-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; dim: 55401 208 </span></span>
<span id="cb12-24"><a href="smokingmouse-rse.html#cb12-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metadata(1): Obtained_from</span></span>
<span id="cb12-25"><a href="smokingmouse-rse.html#cb12-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span id="cb12-26"><a href="smokingmouse-rse.html#cb12-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rownames(55401): ENSMUSG00000102693.1 ENSMUSG00000064842.1 ... ENSMUSG00000064371.1 ENSMUSG00000064372.1</span></span>
<span id="cb12-27"><a href="smokingmouse-rse.html#cb12-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rowData names(13): Length gencodeID ... DE_in_pup_brain_nicotine DE_in_pup_brain_smoking</span></span>
<span id="cb12-28"><a href="smokingmouse-rse.html#cb12-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colnames: NULL</span></span>
<span id="cb12-29"><a href="smokingmouse-rse.html#cb12-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colData names(71): SAMPLE_ID FQCbasicStats ... retained_after_QC_sample_filtering</span></span>
<span id="cb12-30"><a href="smokingmouse-rse.html#cb12-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   retained_after_manual_sample_filtering</span></span></code></pre></div>
</div>
<div id="data-overview" class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Data overview<a href="smokingmouse-rse.html#data-overview" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="assays" class="section level3 hasAnchor" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Assays<a href="smokingmouse-rse.html#assays" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The dataset <code>rse_gene</code> contains the following assays:</p>
<ul>
<li><code>counts</code>: original read counts of the 55,401 mouse genes across 208 samples (inlcuding the 65 nicotine samples of interest).</li>
<li><code>logcounts</code>: normalized and scaled counts (<span class="math inline">\(log_2(CPM + 0.5)\)</span>) of the same genes across the same samples; normalization was carried out applying TMM method with <code>cpm(calcNormFactors())</code> of <em><a href="https://bioconductor.org/packages/3.17/edgeR">edgeR</a></em>.</li>
</ul>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="smokingmouse-rse.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Explore main assay (of raw counts)</span></span>
<span id="cb13-2"><a href="smokingmouse-rse.html#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(rse_gene)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>] <span class="do">## counts for first 3 genes and 3 samples</span></span>
<span id="cb13-3"><a href="smokingmouse-rse.html#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                      [,1] [,2] [,3]</span></span>
<span id="cb13-4"><a href="smokingmouse-rse.html#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000102693.1    0    0    0</span></span>
<span id="cb13-5"><a href="smokingmouse-rse.html#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000064842.1    0    0    0</span></span>
<span id="cb13-6"><a href="smokingmouse-rse.html#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051951.5  811  710  812</span></span>
<span id="cb13-7"><a href="smokingmouse-rse.html#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Access the same raw data with assays()</span></span>
<span id="cb13-8"><a href="smokingmouse-rse.html#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="fu">assays</span>(rse_gene)<span class="sc">$</span>counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb13-9"><a href="smokingmouse-rse.html#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                      [,1] [,2] [,3]</span></span>
<span id="cb13-10"><a href="smokingmouse-rse.html#cb13-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000102693.1    0    0    0</span></span>
<span id="cb13-11"><a href="smokingmouse-rse.html#cb13-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000064842.1    0    0    0</span></span>
<span id="cb13-12"><a href="smokingmouse-rse.html#cb13-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051951.5  811  710  812</span></span>
<span id="cb13-13"><a href="smokingmouse-rse.html#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Access lognorm counts</span></span>
<span id="cb13-14"><a href="smokingmouse-rse.html#cb13-14" aria-hidden="true" tabindex="-1"></a><span class="fu">assays</span>(rse_gene)<span class="sc">$</span>logcounts[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb13-15"><a href="smokingmouse-rse.html#cb13-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                           [,1]      [,2]      [,3]</span></span>
<span id="cb13-16"><a href="smokingmouse-rse.html#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000102693.1 -5.985331 -5.985331 -5.985331</span></span>
<span id="cb13-17"><a href="smokingmouse-rse.html#cb13-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000064842.1 -5.985331 -5.985331 -5.985331</span></span>
<span id="cb13-18"><a href="smokingmouse-rse.html#cb13-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051951.5  4.509114  4.865612  4.944597</span></span></code></pre></div>
</div>
<div id="sample-data" class="section level3 hasAnchor" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Sample data<a href="smokingmouse-rse.html#sample-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><mark style= "background-color: #FCF3CF"> Yellow </mark> variables correspond to SPEAQeasy outputs that are going to be used in downstream analyses.</li>
<li><mark style= "background-color: #FAECF8"> Pink </mark> variables are specific of the study, such as sample metadata and some others containing additional information of the genes.</li>
<li><mark style= "background-color: #DFF0FE"> Blue </mark> variables are Quality-Control metrics computed by <code>addPerCellQC()</code> of <em><a href="https://bioconductor.org/packages/3.17/scuttle">scuttle</a></em>.</li>
</ul>
<p>The same RSE contains the sample information in <code>colData(RSE)</code>:</p>
<ul>
<li><mark style= "background-color: #FCF3CF"> <span style="font-family: monospace"> SAMPLE_ID</span> </mark>: is the name of the sample.</li>
<li><mark style= "background-color: #FCF3CF"> <span style="font-family: monospace"> ERCCsumLogErr</span> </mark>: a summary statistic quantifying overall difference of expected and actual ERCC concentrations for one sample.</li>
<li><mark style= "background-color: #FCF3CF"> <span style="font-family: monospace"> overallMapRate</span> </mark>: the decimal fraction of reads which successfully mapped to the reference genome (i.e. <em>numMapped</em> / <em>numReads</em>).</li>
<li><mark style= "background-color: #FCF3CF"> <span style="font-family: monospace"> mitoMapped</span> </mark>: the number of reads which successfully mapped to the mitochondrial chromosome.</li>
<li><mark style= "background-color: #FCF3CF"> <span style="font-family: monospace"> totalMapped</span> </mark>: the number of reads which successfully mapped to the canonical sequences in the reference genome (excluding mitochondrial chromosomes).</li>
<li><mark style= "background-color: #FCF3CF"> <span style="font-family: monospace"> mitoRate</span> </mark>: the decimal fraction of reads which mapped to the mitochondrial chromosome, of those which map at all (i.e. <em>mitoMapped</em> / (<em>totalMapped</em> + <em>mitoMapped</em>))</li>
<li><mark style= "background-color: #FCF3CF"> <span style="font-family: monospace"> totalAssignedGene</span> </mark>: the decimal fraction of reads assigned unambiguously to a gene (including mitochondrial genes), with <code>featureCounts</code> (Liao et al. 2014), of those in total.</li>
<li><mark style= "background-color: #FCF3CF"> <span style="font-family: monospace"> rRNA_rate</span> </mark>: the decimal fraction of reads assigned to a gene whose type is ‘rRNA’, of those assigned to any gene.</li>
<li><mark style= "background-color: #FAECF8"> <span style="font-family: monospace"> Tissue</span> </mark>: tissue (mouse brain or blood) from which the sample comes.</li>
<li><mark style= "background-color: #FAECF8"> <span style="font-family: monospace"> Age</span> </mark>: if the sample comes from an adult or a pup mouse.</li>
<li><mark style= "background-color: #FAECF8"> <span style="font-family: monospace"> Sex</span> </mark>: if the sample comes from a female (F) or male (M) mouse.</li>
<li><mark style= "background-color: #FAECF8"> <span style="font-family: monospace"> Expt</span> </mark>: the experiment (nicotine or smoking exposure) to which the sample mouse was subjected; it could be an exposed or a control mouse of that experiment.</li>
<li><mark style= "background-color: #FAECF8"> <span style="font-family: monospace"> Group</span> </mark>: if the sample belongs to a nicotine/smoking-exposed mouse (Expt) or a nicotine/smoking control mouse (Ctrl).</li>
<li><mark style= "background-color: #FAECF8"> <span style="font-family: monospace"> plate</span> </mark>: is the plate (1,2 or 3) in which the sample library was prepared.</li>
<li><mark style= "background-color: #FAECF8"> <span style="font-family: monospace"> Pregnancy</span> </mark>: if the sample comes from a pregnant (Yes) or not pregnant (No) mouse.</li>
<li><mark style= "background-color: #FAECF8"> <span style="font-family: monospace"> medium</span> </mark>: is the medium in which the sample was treated: water for brain samples and an elution buffer (EB) for the blood ones.</li>
<li><mark style= "background-color: #FAECF8"> <span style="font-family: monospace"> flowcell</span> </mark>: is the sequencing batch of each sample.</li>
<li><mark style= "background-color: #DFF0FE"> <span style="font-family: monospace"> sum</span> </mark>: library size (total sum of counts across all genes for each sample).</li>
<li><mark style= "background-color: #DFF0FE"> <span style="font-family: monospace"> detected</span> </mark>: number of non-zero expressed genes in each sample.</li>
<li><mark style= "background-color: #DFF0FE"> <span style="font-family: monospace"> subsets_Mito_sum</span> </mark>: total sum of read counts of mt genes in each sample.</li>
<li><mark style= "background-color: #DFF0FE"> <span style="font-family: monospace"> subsets_Mito_detected</span> </mark>: total number of mt genes in each sample.</li>
<li><mark style= "background-color: #DFF0FE"> <span style="font-family: monospace"> subsets_Mito_percent</span> </mark>: % of mt genes’ counts of the total counts of the sample.</li>
<li><mark style= "background-color: #DFF0FE"> <span style="font-family: monospace"> subsets_Ribo_sum</span> </mark>: total sum of read counts of ribosomal genes in each sample.</li>
<li><mark style= "background-color: #DFF0FE"> <span style="font-family: monospace"> subsets_Ribo_detected</span> </mark>: total number of ribosomal genes in each sample.</li>
<li><mark style= "background-color: #DFF0FE"> <span style="font-family: monospace"> subsets_Ribo_percent</span> </mark>: % of ribosomal genes’ counts of the total counts of the sample.</li>
</ul>
<div class="alert alert-info">
<p><strong>Note:</strong> in our case, we’ll use samples from the nicotine experiment only, so all samples come from brain and were treated in water.</p>
</div>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="smokingmouse-rse.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Data for first 3 samples and 5 variables</span></span>
<span id="cb14-2"><a href="smokingmouse-rse.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(rse_gene)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb14-3"><a href="smokingmouse-rse.html#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; DataFrame with 3 rows and 5 columns</span></span>
<span id="cb14-4"><a href="smokingmouse-rse.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     SAMPLE_ID FQCbasicStats perBaseQual perTileQual  perSeqQual</span></span>
<span id="cb14-5"><a href="smokingmouse-rse.html#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   &lt;character&gt;   &lt;character&gt; &lt;character&gt; &lt;character&gt; &lt;character&gt;</span></span>
<span id="cb14-6"><a href="smokingmouse-rse.html#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1 Sample_2914          PASS        PASS        PASS        PASS</span></span>
<span id="cb14-7"><a href="smokingmouse-rse.html#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2 Sample_4042          PASS        PASS        PASS        PASS</span></span>
<span id="cb14-8"><a href="smokingmouse-rse.html#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3 Sample_4043          PASS        PASS        PASS        PASS</span></span></code></pre></div>
</div>
<div id="gene-information" class="section level3 hasAnchor" number="4.2.3">
<h3><span class="header-section-number">4.2.3</span> Gene Information<a href="smokingmouse-rse.html#gene-information" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Among the information in <code>rowData(RSE)</code> the next variables are of interest for the analysis:</p>
<ul>
<li><mark style= "background-color: #FCF3CF"> <span style="font-family: monospace"> gencodeID</span> </mark>: GENCODE ID of each gene.</li>
<li><mark style= "background-color: #FCF3CF"> <span style="font-family: monospace"> ensemblID</span> </mark>: gene ID in Ensembl database.</li>
<li><mark style= "background-color: #FCF3CF"> <span style="font-family: monospace"> EntrezID</span> </mark>: identifier of each gene in NCBI Entrez database.</li>
<li><mark style= "background-color: #FCF3CF"> <span style="font-family: monospace"> Symbol</span> </mark>: official gene symbol for each mouse gene.</li>
<li><mark style= "background-color: #FAECF8"> <span style="font-family: monospace"> retained_after_feature_filtering</span> </mark>: Boolean variable that equals TRUE if the gene passed the gene filtering (with <code>filterByExpr()</code> of <em><a href="https://bioconductor.org/packages/3.17/edgeR">edgeR</a></em>) based on its expression levels and FALSE if not.</li>
<li><mark style= "background-color: #FAECF8"> <span style="font-family: monospace"> DE_in_adult_brain_nicotine</span> </mark>: Boolean variable that equals TRUE if the feature is differentially expressed (DE) in adult brain samples exposed to nicotine and FALSE if not.</li>
<li><mark style= "background-color: #FAECF8"> <span style="font-family: monospace"> DE_in_pup_brain_nicotine</span> </mark>: Boolean variable that equals TRUE if the feature is differentially expressed (DE) in pup brain samples exposed to nicotine and FALSE if not.</li>
</ul>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="smokingmouse-rse.html#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Data for first 3 genes and 5 variables</span></span>
<span id="cb15-2"><a href="smokingmouse-rse.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(rse_gene)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb15-3"><a href="smokingmouse-rse.html#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; DataFrame with 3 rows and 5 columns</span></span>
<span id="cb15-4"><a href="smokingmouse-rse.html#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                         Length            gencodeID          ensemblID      gene_type    EntrezID</span></span>
<span id="cb15-5"><a href="smokingmouse-rse.html#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                      &lt;integer&gt;          &lt;character&gt;        &lt;character&gt;    &lt;character&gt; &lt;character&gt;</span></span>
<span id="cb15-6"><a href="smokingmouse-rse.html#cb15-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000102693.1      1070 ENSMUSG00000102693.1 ENSMUSG00000102693            TEC       71042</span></span>
<span id="cb15-7"><a href="smokingmouse-rse.html#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000064842.1       110 ENSMUSG00000064842.1 ENSMUSG00000064842          snRNA          NA</span></span>
<span id="cb15-8"><a href="smokingmouse-rse.html#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSMUSG00000051951.5      6094 ENSMUSG00000051951.5 ENSMUSG00000051951 protein_coding      497097</span></span></code></pre></div>
<style>
p.exercise  {
background-color: #E4EDE2;
padding: 9px;
border: 1px solid black;
border-radius: 10px;
font-family: sans-serif;
}

</style>
<p class="exercise">
📑 <strong>Exercise 1</strong>: How would you access data of a specific sample variable?
</p>
<p>For illustrative purposes, we’ll use nicotine data at the gene level only so let’s access that data.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="smokingmouse-rse.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Original dimensions of the data</span></span>
<span id="cb16-2"><a href="smokingmouse-rse.html#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(rse_gene)</span>
<span id="cb16-3"><a href="smokingmouse-rse.html#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 55401   208</span></span>
<span id="cb16-4"><a href="smokingmouse-rse.html#cb16-4" aria-hidden="true" tabindex="-1"></a>rse_gene_nic <span class="ot">&lt;-</span> rse_gene[,<span class="fu">which</span>(rse_gene<span class="sc">$</span>Expt<span class="sc">==</span><span class="st">&#39;Nicotine&#39;</span>)]</span>
<span id="cb16-5"><a href="smokingmouse-rse.html#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="do">## New dimensions</span></span>
<span id="cb16-6"><a href="smokingmouse-rse.html#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(rse_gene_nic)</span>
<span id="cb16-7"><a href="smokingmouse-rse.html#cb16-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 55401    65</span></span></code></pre></div>
<p class="exercise">
📑 <strong>Exercise 2</strong>: How could you check that all samples are from the nicotine experiment?
</p>
<p class="exercise">
📑 <strong>Exercise 3</strong>: How many nicotine samples correspond to adults and how many to pups? How many pups were males and how many were females?
</p>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  setting  value
#&gt;  version  R version 4.3.0 (2023-04-21)
#&gt;  os       Ubuntu 22.04.2 LTS
#&gt;  system   x86_64, linux-gnu
#&gt;  ui       X11
#&gt;  language (EN)
#&gt;  collate  en_US.UTF-8
#&gt;  ctype    en_US.UTF-8
#&gt;  tz       UTC
#&gt;  date     2023-07-02
#&gt;  pandoc   2.19.2 @ /usr/local/bin/ (via rmarkdown)
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  package                * version   date (UTC) lib source
#&gt;  airway                 * 1.20.0    2023-04-27 [1] Bioconductor
#&gt;  AnnotationDbi            1.62.1    2023-05-02 [1] Bioconductor
#&gt;  AnnotationHub            3.8.0     2023-04-25 [1] Bioconductor
#&gt;  aod                      1.3.2     2022-04-02 [1] RSPM (R 4.3.0)
#&gt;  attempt                  0.3.1     2020-05-03 [1] RSPM (R 4.3.0)
#&gt;  backports                1.4.1     2021-12-13 [1] RSPM (R 4.3.0)
#&gt;  base64enc                0.1-3     2015-07-28 [2] RSPM (R 4.3.0)
#&gt;  beachmat                 2.16.0    2023-04-25 [1] Bioconductor
#&gt;  beeswarm                 0.4.0     2021-06-01 [1] RSPM (R 4.3.0)
#&gt;  benchmarkme              1.0.8     2022-06-12 [1] RSPM (R 4.3.0)
#&gt;  benchmarkmeData          1.0.4     2020-04-23 [1] RSPM (R 4.3.0)
#&gt;  bibtex                   0.5.1     2023-01-26 [1] RSPM (R 4.3.0)
#&gt;  Biobase                * 2.60.0    2023-04-25 [1] Bioconductor
#&gt;  BiocFileCache          * 2.8.0     2023-04-25 [1] Bioconductor
#&gt;  BiocGenerics           * 0.46.0    2023-04-25 [1] Bioconductor
#&gt;  BiocIO                   1.10.0    2023-04-25 [1] Bioconductor
#&gt;  BiocManager              1.30.21   2023-06-10 [1] RSPM (R 4.3.0)
#&gt;  BiocNeighbors            1.18.0    2023-04-25 [1] Bioconductor
#&gt;  BiocParallel           * 1.34.2    2023-05-22 [1] Bioconductor
#&gt;  BiocSingular             1.16.0    2023-04-25 [1] Bioconductor
#&gt;  BiocStyle              * 2.28.0    2023-04-25 [1] Bioconductor
#&gt;  biocthis               * 1.10.3    2023-06-07 [1] Bioconductor
#&gt;  BiocVersion              3.17.1    2022-11-04 [2] Bioconductor
#&gt;  Biostrings               2.68.1    2023-05-16 [1] Bioconductor
#&gt;  bit                      4.0.5     2022-11-15 [1] RSPM (R 4.3.0)
#&gt;  bit64                    4.0.5     2020-08-30 [1] RSPM (R 4.3.0)
#&gt;  bitops                   1.0-7     2021-04-24 [1] RSPM (R 4.3.0)
#&gt;  blob                     1.2.4     2023-03-17 [1] RSPM (R 4.3.0)
#&gt;  bookdown                 0.34      2023-05-09 [1] RSPM (R 4.3.0)
#&gt;  boot                     1.3-28.1  2022-11-22 [3] CRAN (R 4.3.0)
#&gt;  broom                    1.0.5     2023-06-09 [1] RSPM (R 4.3.0)
#&gt;  bslib                    0.5.0     2023-06-09 [2] RSPM (R 4.3.0)
#&gt;  cachem                   1.0.8     2023-05-01 [2] RSPM (R 4.3.0)
#&gt;  caTools                  1.18.2    2021-03-28 [1] RSPM (R 4.3.0)
#&gt;  checkmate                2.2.0     2023-04-27 [1] RSPM (R 4.3.0)
#&gt;  circlize                 0.4.15    2022-05-10 [1] RSPM (R 4.3.0)
#&gt;  cli                      3.6.1     2023-03-23 [2] RSPM (R 4.3.0)
#&gt;  clue                     0.3-64    2023-01-31 [1] RSPM (R 4.3.0)
#&gt;  cluster                  2.1.4     2022-08-22 [3] CRAN (R 4.3.0)
#&gt;  codetools                0.2-19    2023-02-01 [3] CRAN (R 4.3.0)
#&gt;  colorspace               2.1-0     2023-01-23 [1] RSPM (R 4.3.0)
#&gt;  colourpicker             1.2.0     2022-10-28 [1] RSPM (R 4.3.0)
#&gt;  ComplexHeatmap         * 2.16.0    2023-04-25 [1] Bioconductor
#&gt;  config                   0.3.1     2020-12-17 [1] RSPM (R 4.3.0)
#&gt;  cowplot                * 1.1.1     2020-12-30 [1] RSPM (R 4.3.0)
#&gt;  crayon                   1.5.2     2022-09-29 [2] RSPM (R 4.3.0)
#&gt;  curl                     5.0.1     2023-06-07 [2] RSPM (R 4.3.0)
#&gt;  data.table               1.14.8    2023-02-17 [1] RSPM (R 4.3.0)
#&gt;  DBI                      1.1.3     2022-06-18 [1] RSPM (R 4.3.0)
#&gt;  dbplyr                 * 2.3.2     2023-03-21 [1] RSPM (R 4.3.0)
#&gt;  DelayedArray             0.26.5    2023-06-30 [1] Bioconductor
#&gt;  DelayedMatrixStats       1.22.1    2023-06-09 [1] Bioconductor
#&gt;  digest                   0.6.32    2023-06-26 [2] RSPM (R 4.3.0)
#&gt;  doParallel               1.0.17    2022-02-07 [1] RSPM (R 4.3.0)
#&gt;  dotCall64                1.0-2     2022-10-03 [1] RSPM (R 4.3.0)
#&gt;  dplyr                    1.1.2     2023-04-20 [1] RSPM (R 4.3.0)
#&gt;  dqrng                    0.3.0     2021-05-01 [1] RSPM (R 4.3.0)
#&gt;  DropletUtils             1.20.0    2023-04-25 [1] Bioconductor
#&gt;  DT                       0.28      2023-05-18 [1] RSPM (R 4.3.0)
#&gt;  edgeR                  * 3.42.4    2023-05-31 [1] Bioconductor
#&gt;  ellipsis                 0.3.2     2021-04-29 [2] RSPM (R 4.3.0)
#&gt;  EnvStats                 2.7.0     2022-03-07 [1] RSPM (R 4.3.0)
#&gt;  evaluate                 0.21      2023-05-05 [2] RSPM (R 4.3.0)
#&gt;  ExperimentHub            2.8.0     2023-04-25 [1] Bioconductor
#&gt;  ExploreModelMatrix     * 1.12.0    2023-04-25 [1] Bioconductor
#&gt;  fansi                    1.0.4     2023-01-22 [2] RSPM (R 4.3.0)
#&gt;  fastmap                  1.1.1     2023-02-24 [2] RSPM (R 4.3.0)
#&gt;  fields                   14.1      2022-08-12 [1] RSPM (R 4.3.0)
#&gt;  filelock                 1.0.2     2018-10-05 [1] RSPM (R 4.3.0)
#&gt;  foreach                  1.5.2     2022-02-02 [1] RSPM (R 4.3.0)
#&gt;  foreign                  0.8-84    2022-12-06 [3] CRAN (R 4.3.0)
#&gt;  Formula                  1.2-5     2023-02-24 [1] RSPM (R 4.3.0)
#&gt;  fs                       1.6.2     2023-04-25 [2] RSPM (R 4.3.0)
#&gt;  generics                 0.1.3     2022-07-05 [1] RSPM (R 4.3.0)
#&gt;  GenomeInfoDb           * 1.36.1    2023-06-21 [1] Bioconductor
#&gt;  GenomeInfoDbData         1.2.10    2023-05-17 [1] Bioconductor
#&gt;  GenomicAlignments        1.36.0    2023-04-25 [1] Bioconductor
#&gt;  GenomicRanges          * 1.52.0    2023-04-25 [1] Bioconductor
#&gt;  GetoptLong               1.0.5     2020-12-15 [1] RSPM (R 4.3.0)
#&gt;  ggbeeswarm               0.7.2     2023-04-29 [1] RSPM (R 4.3.0)
#&gt;  ggplot2                * 3.4.2     2023-04-03 [1] RSPM (R 4.3.0)
#&gt;  ggrepel                * 0.9.3     2023-02-03 [1] RSPM (R 4.3.0)
#&gt;  GlobalOptions            0.1.2     2020-06-10 [1] RSPM (R 4.3.0)
#&gt;  glue                     1.6.2     2022-02-24 [2] RSPM (R 4.3.0)
#&gt;  golem                    0.4.1     2023-06-05 [1] RSPM (R 4.3.0)
#&gt;  gplots                   3.1.3     2022-04-25 [1] RSPM (R 4.3.0)
#&gt;  gridExtra                2.3       2017-09-09 [1] RSPM (R 4.3.0)
#&gt;  gtable                   0.3.3     2023-03-21 [1] RSPM (R 4.3.0)
#&gt;  gtools                   3.9.4     2022-11-27 [1] RSPM (R 4.3.0)
#&gt;  HDF5Array                1.28.1    2023-05-01 [1] Bioconductor
#&gt;  here                   * 1.0.1     2020-12-13 [1] RSPM (R 4.3.0)
#&gt;  Hmisc                  * 5.1-0     2023-05-08 [1] RSPM (R 4.3.0)
#&gt;  hms                      1.1.3     2023-03-21 [1] RSPM (R 4.3.0)
#&gt;  htmlTable                2.4.1     2022-07-07 [1] RSPM (R 4.3.0)
#&gt;  htmltools                0.5.5     2023-03-23 [2] RSPM (R 4.3.0)
#&gt;  htmlwidgets              1.6.2     2023-03-17 [2] RSPM (R 4.3.0)
#&gt;  httpuv                   1.6.11    2023-05-11 [2] RSPM (R 4.3.0)
#&gt;  httr                     1.4.6     2023-05-08 [2] RSPM (R 4.3.0)
#&gt;  igraph                   1.5.0     2023-06-16 [1] RSPM (R 4.3.0)
#&gt;  interactiveDisplayBase   1.38.0    2023-04-25 [1] Bioconductor
#&gt;  IRanges                * 2.34.1    2023-06-22 [1] Bioconductor
#&gt;  irlba                    2.3.5.1   2022-10-03 [1] RSPM (R 4.3.0)
#&gt;  iSEE                   * 2.12.0    2023-04-25 [1] Bioconductor
#&gt;  iterators                1.0.14    2022-02-05 [1] RSPM (R 4.3.0)
#&gt;  jquerylib                0.1.4     2021-04-26 [2] RSPM (R 4.3.0)
#&gt;  jsonlite                 1.8.7     2023-06-29 [2] RSPM (R 4.3.0)
#&gt;  KEGGREST                 1.40.0    2023-04-25 [1] Bioconductor
#&gt;  KernSmooth               2.23-21   2023-05-03 [3] RSPM (R 4.3.0)
#&gt;  knitr                    1.43      2023-05-25 [2] RSPM (R 4.3.0)
#&gt;  later                    1.3.1     2023-05-02 [2] RSPM (R 4.3.0)
#&gt;  lattice                  0.21-8    2023-04-05 [3] CRAN (R 4.3.0)
#&gt;  lazyeval                 0.2.2     2019-03-15 [1] RSPM (R 4.3.0)
#&gt;  lifecycle                1.0.3     2022-10-07 [2] RSPM (R 4.3.0)
#&gt;  limma                  * 3.56.2    2023-06-04 [1] Bioconductor
#&gt;  lme4                     1.1-33    2023-04-25 [1] RSPM (R 4.3.0)
#&gt;  lobstr                 * 1.1.2     2022-06-22 [1] RSPM (R 4.3.0)
#&gt;  locfit                   1.5-9.8   2023-06-11 [1] RSPM (R 4.3.0)
#&gt;  lubridate                1.9.2     2023-02-10 [1] RSPM (R 4.3.0)
#&gt;  magick                   2.7.4     2023-03-09 [1] RSPM (R 4.3.0)
#&gt;  magrittr                 2.0.3     2022-03-30 [2] RSPM (R 4.3.0)
#&gt;  maps                     3.4.1     2022-10-30 [1] RSPM (R 4.3.0)
#&gt;  MASS                     7.3-60    2023-05-04 [3] RSPM (R 4.3.0)
#&gt;  Matrix                   1.5-4.1   2023-05-18 [3] RSPM (R 4.3.0)
#&gt;  MatrixGenerics         * 1.12.2    2023-06-09 [1] Bioconductor
#&gt;  matrixStats            * 1.0.0     2023-06-02 [1] RSPM (R 4.3.0)
#&gt;  memoise                  2.0.1     2021-11-26 [2] RSPM (R 4.3.0)
#&gt;  mgcv                     1.8-42    2023-03-02 [3] CRAN (R 4.3.0)
#&gt;  mime                     0.12      2021-09-28 [2] RSPM (R 4.3.0)
#&gt;  miniUI                   0.1.1.1   2018-05-18 [2] RSPM (R 4.3.0)
#&gt;  minqa                    1.2.5     2022-10-19 [1] RSPM (R 4.3.0)
#&gt;  munsell                  0.5.0     2018-06-12 [1] RSPM (R 4.3.0)
#&gt;  mvtnorm                  1.2-2     2023-06-08 [1] RSPM (R 4.3.0)
#&gt;  nlme                     3.1-162   2023-01-31 [3] CRAN (R 4.3.0)
#&gt;  nloptr                   2.0.3     2022-05-26 [1] RSPM (R 4.3.0)
#&gt;  nnet                     7.3-19    2023-05-03 [3] RSPM (R 4.3.0)
#&gt;  paletteer                1.5.0     2022-10-19 [1] RSPM (R 4.3.0)
#&gt;  patchwork              * 1.1.2     2022-08-19 [1] RSPM (R 4.3.0)
#&gt;  pbkrtest                 0.5.2     2023-01-19 [1] RSPM (R 4.3.0)
#&gt;  pheatmap               * 1.0.12    2019-01-04 [1] RSPM (R 4.3.0)
#&gt;  pillar                   1.9.0     2023-03-22 [2] RSPM (R 4.3.0)
#&gt;  pkgconfig                2.0.3     2019-09-22 [2] RSPM (R 4.3.0)
#&gt;  plotly                   4.10.2    2023-06-03 [1] RSPM (R 4.3.0)
#&gt;  plyr                     1.8.8     2022-11-11 [1] RSPM (R 4.3.0)
#&gt;  png                      0.1-8     2022-11-29 [1] RSPM (R 4.3.0)
#&gt;  postcards              * 0.2.3     2022-01-07 [1] RSPM (R 4.3.0)
#&gt;  prettyunits              1.1.1     2020-01-24 [2] RSPM (R 4.3.0)
#&gt;  progress                 1.2.2     2019-05-16 [1] RSPM (R 4.3.0)
#&gt;  promises                 1.2.0.1   2021-02-11 [2] RSPM (R 4.3.0)
#&gt;  purrr                    1.0.1     2023-01-10 [2] RSPM (R 4.3.0)
#&gt;  R.cache                  0.16.0    2022-07-21 [1] RSPM (R 4.3.0)
#&gt;  R.methodsS3              1.8.2     2022-06-13 [1] RSPM (R 4.3.0)
#&gt;  R.oo                     1.25.0    2022-06-12 [1] RSPM (R 4.3.0)
#&gt;  R.utils                  2.12.2    2022-11-11 [1] RSPM (R 4.3.0)
#&gt;  R6                       2.5.1     2021-08-19 [2] RSPM (R 4.3.0)
#&gt;  rappdirs                 0.3.3     2021-01-31 [2] RSPM (R 4.3.0)
#&gt;  rbibutils                2.2.13    2023-01-13 [1] RSPM (R 4.3.0)
#&gt;  RColorBrewer           * 1.1-3     2022-04-03 [1] RSPM (R 4.3.0)
#&gt;  Rcpp                     1.0.10    2023-01-22 [2] RSPM (R 4.3.0)
#&gt;  RCurl                    1.98-1.12 2023-03-27 [1] RSPM (R 4.3.0)
#&gt;  Rdpack                   2.4       2022-07-20 [1] RSPM (R 4.3.0)
#&gt;  recount3               * 1.10.2    2023-05-07 [1] Bioconductor
#&gt;  RefManageR             * 1.4.0     2022-09-30 [1] RSPM (R 4.3.0)
#&gt;  remaCor                  0.0.16    2023-06-21 [1] RSPM (R 4.3.0)
#&gt;  rematch2                 2.1.2     2020-05-01 [2] RSPM (R 4.3.0)
#&gt;  reshape2                 1.4.4     2020-04-09 [1] RSPM (R 4.3.0)
#&gt;  restfulr                 0.0.15    2022-06-16 [1] RSPM (R 4.3.0)
#&gt;  rhdf5                    2.44.0    2023-04-25 [1] Bioconductor
#&gt;  rhdf5filters             1.12.1    2023-04-30 [1] Bioconductor
#&gt;  Rhdf5lib                 1.22.0    2023-04-25 [1] Bioconductor
#&gt;  RhpcBLASctl              0.23-42   2023-02-11 [1] RSPM (R 4.3.0)
#&gt;  rintrojs                 0.3.2     2022-08-09 [1] RSPM (R 4.3.0)
#&gt;  rjson                    0.2.21    2022-01-09 [1] RSPM (R 4.3.0)
#&gt;  rlang                  * 1.1.1     2023-04-28 [2] RSPM (R 4.3.0)
#&gt;  rmarkdown                2.22      2023-06-01 [2] RSPM (R 4.3.0)
#&gt;  rpart                    4.1.19    2022-10-21 [3] CRAN (R 4.3.0)
#&gt;  rprojroot                2.0.3     2022-04-02 [2] RSPM (R 4.3.0)
#&gt;  Rsamtools                2.16.0    2023-04-25 [1] Bioconductor
#&gt;  RSQLite                  2.3.1     2023-04-03 [1] RSPM (R 4.3.0)
#&gt;  rstudioapi               0.14      2022-08-22 [2] RSPM (R 4.3.0)
#&gt;  rsvd                     1.0.5     2021-04-16 [1] RSPM (R 4.3.0)
#&gt;  rtracklayer              1.60.0    2023-04-25 [1] Bioconductor
#&gt;  S4Arrays                 1.0.4     2023-05-14 [1] Bioconductor
#&gt;  S4Vectors              * 0.38.1    2023-05-02 [1] Bioconductor
#&gt;  sass                     0.4.6     2023-05-03 [2] RSPM (R 4.3.0)
#&gt;  ScaledMatrix             1.8.1     2023-05-03 [1] Bioconductor
#&gt;  scales                   1.2.1     2022-08-20 [1] RSPM (R 4.3.0)
#&gt;  scater                 * 1.28.0    2023-04-25 [1] Bioconductor
#&gt;  scuttle                * 1.10.1    2023-05-02 [1] Bioconductor
#&gt;  sessioninfo            * 1.2.2     2021-12-06 [2] RSPM (R 4.3.0)
#&gt;  shape                    1.4.6     2021-05-19 [1] RSPM (R 4.3.0)
#&gt;  shiny                    1.7.4     2022-12-15 [2] RSPM (R 4.3.0)
#&gt;  shinyAce                 0.4.2     2022-05-06 [1] RSPM (R 4.3.0)
#&gt;  shinydashboard           0.7.2     2021-09-30 [1] RSPM (R 4.3.0)
#&gt;  shinyjs                  2.1.0     2021-12-23 [1] RSPM (R 4.3.0)
#&gt;  shinyWidgets             0.7.6     2023-01-08 [1] RSPM (R 4.3.0)
#&gt;  SingleCellExperiment   * 1.22.0    2023-04-25 [1] Bioconductor
#&gt;  smokingMouse           * 0.99.5    2023-06-28 [1] Github (LieberInstitute/smokingMouse@2e7640c)
#&gt;  spam                     2.9-1     2022-08-07 [1] RSPM (R 4.3.0)
#&gt;  sparseMatrixStats        1.12.1    2023-06-20 [1] Bioconductor
#&gt;  SpatialExperiment      * 1.10.0    2023-04-25 [1] Bioconductor
#&gt;  spatialLIBD            * 1.12.0    2023-04-27 [1] Bioconductor
#&gt;  statmod                  1.5.0     2023-01-06 [1] RSPM (R 4.3.0)
#&gt;  stringi                  1.7.12    2023-01-11 [2] RSPM (R 4.3.0)
#&gt;  stringr                * 1.5.0     2022-12-02 [2] RSPM (R 4.3.0)
#&gt;  styler                   1.10.1    2023-06-05 [1] RSPM (R 4.3.0)
#&gt;  SummarizedExperiment   * 1.30.2    2023-06-06 [1] Bioconductor
#&gt;  tibble                   3.2.1     2023-03-20 [2] RSPM (R 4.3.0)
#&gt;  tidyr                    1.3.0     2023-01-24 [1] RSPM (R 4.3.0)
#&gt;  tidyselect               1.2.0     2022-10-10 [1] RSPM (R 4.3.0)
#&gt;  timechange               0.2.0     2023-01-11 [1] RSPM (R 4.3.0)
#&gt;  usethis                * 2.2.1     2023-06-23 [2] RSPM (R 4.3.0)
#&gt;  utf8                     1.2.3     2023-01-31 [2] RSPM (R 4.3.0)
#&gt;  variancePartition      * 1.30.2    2023-06-07 [1] Bioconductor
#&gt;  vctrs                    0.6.3     2023-06-14 [2] RSPM (R 4.3.0)
#&gt;  vipor                    0.4.5     2017-03-22 [1] RSPM (R 4.3.0)
#&gt;  viridis                  0.6.3     2023-05-03 [1] RSPM (R 4.3.0)
#&gt;  viridisLite              0.4.2     2023-05-02 [1] RSPM (R 4.3.0)
#&gt;  withr                    2.5.0     2022-03-03 [2] RSPM (R 4.3.0)
#&gt;  xfun                     0.39      2023-04-20 [2] RSPM (R 4.3.0)
#&gt;  XML                      3.99-0.14 2023-03-19 [1] RSPM (R 4.3.0)
#&gt;  xml2                     1.3.4     2023-04-27 [2] RSPM (R 4.3.0)
#&gt;  xtable                   1.8-4     2019-04-21 [2] RSPM (R 4.3.0)
#&gt;  XVector                  0.40.0    2023-04-25 [1] Bioconductor
#&gt;  yaml                     2.3.7     2023-01-23 [2] RSPM (R 4.3.0)
#&gt;  zlibbioc                 1.46.0    2023-04-25 [1] Bioconductor
#&gt; 
#&gt;  [1] /__w/_temp/Library
#&gt;  [2] /usr/local/lib/R/site-library
#&gt;  [3] /usr/local/lib/R/library
#&gt; 
#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
</div>
<div class="container">
    <footer class="site-footer">

      <center>
<p class="powered-by">
© 2011-2023. All thoughts and opinions here are my own. The icon was designed by <a href="https://www.linkedin.com/in/mauricio-guzman-6529b551/">Mauricio Guzmán</a> and is inspired by <a href="https://en.wikipedia.org/wiki/Huichol">Huichol culture</a>; it represents my community building interests.
</p>

<p class="powered-by copyright-license-text">
This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">CC BY NC SA 4.0</a>
</p>

<p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">
      <img src="https://mirrors.creativecommons.org/presskit/icons/cc.svg" alt="CC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/by.svg" alt="CC by icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/nc.svg" alt="CC NC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/sa.svg" alt="CC SA icon" width="32px">
    </a>
</p>

<div id="clustrmaps-widget"></div><script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=300&t=m&d=Zo-deincaJGsqhqjIWsujtdoNOMJsnnZLefKmxDKVv8&co=124b73&cmo=dd53ff&cmn=4ec724'></script>

<p class="powered-by">
Published with <a href="https://bookdown.org/" target="_blank" rel="noopener">Bookdown</a>
</p>

</center>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-45299226-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-45299226-1');
</script>

</footer>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="summarizedexperiment-review.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="isee-introduction.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
