<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 SummarizedExperiment review | Statistical Analysis of Genome Scale Data 2023</title>
  <meta name="description" content="3 SummarizedExperiment review | Statistical Analysis of Genome Scale Data 2023" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="3 SummarizedExperiment review | Statistical Analysis of Genome Scale Data 2023" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lcolladotor/cshl_rstats_genome_scale_2023" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 SummarizedExperiment review | Statistical Analysis of Genome Scale Data 2023" />
  
  
  

<meta name="author" content="Leonardo Collado-Torres" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />
<link rel="prev" href="smokingmouse-study.html"/>
<link rel="next" href="smokingmouse-rse.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#code-of-conduct"><i class="fa fa-check"></i>Code of Conduct</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-schedule"><i class="fa fa-check"></i>Course Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#external-links"><i class="fa fa-check"></i>External links</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-prerequisites"><i class="fa fa-check"></i>Course Prerequisites</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html"><i class="fa fa-check"></i><b>1</b> SPEAQeasy introduction</a></li>
<li class="chapter" data-level="2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html"><i class="fa fa-check"></i><b>2</b> smokingMouse study</a>
<ul>
<li class="chapter" data-level="2.1" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#overview-1"><i class="fa fa-check"></i><b>2.2</b> Overview</a></li>
<li class="chapter" data-level="2.3" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#goals"><i class="fa fa-check"></i><b>2.3</b> Goals</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#main-objective"><i class="fa fa-check"></i><b>2.3.1</b> Main objective</a></li>
<li class="chapter" data-level="2.3.2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#secondary-objectives"><i class="fa fa-check"></i><b>2.3.2</b> Secondary objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#study-design"><i class="fa fa-check"></i><b>2.4</b> Study design</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html"><i class="fa fa-check"></i><b>3</b> SummarizedExperiment review</a>
<ul>
<li class="chapter" data-level="3.1" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#overview-2"><i class="fa fa-check"></i><b>3.1</b> Overview</a></li>
<li class="chapter" data-level="3.2" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#quiz"><i class="fa fa-check"></i><b>3.2</b> Quiz</a></li>
<li class="chapter" data-level="3.3" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#exercises"><i class="fa fa-check"></i><b>3.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html"><i class="fa fa-check"></i><b>4</b> smokingMouse RSE</a>
<ul>
<li class="chapter" data-level="4.1" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#download-data"><i class="fa fa-check"></i><b>4.1</b> Download data</a></li>
<li class="chapter" data-level="4.2" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#data-overview"><i class="fa fa-check"></i><b>4.2</b> Data overview</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#assays"><i class="fa fa-check"></i><b>4.2.1</b> Assays</a></li>
<li class="chapter" data-level="4.2.2" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#sample-data"><i class="fa fa-check"></i><b>4.2.2</b> Sample data</a></li>
<li class="chapter" data-level="4.2.3" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#gene-information"><i class="fa fa-check"></i><b>4.2.3</b> Gene Information</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="isee-introduction.html"><a href="isee-introduction.html"><i class="fa fa-check"></i><b>5</b> iSEE introduction</a></li>
<li class="chapter" data-level="6" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html"><i class="fa fa-check"></i><b>6</b> RSE plotting basics with smokingMouse</a>
<ul>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#data-preparation"><i class="fa fa-check"></i>1. Data preparation</a>
<ul>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#data-normalization"><i class="fa fa-check"></i>1.1 Data normalization</a></li>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#gene-filtering"><i class="fa fa-check"></i>1.2 Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#exploratory-data-analysis"><i class="fa fa-check"></i>2. Exploratory Data Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#quality-control-analysis"><i class="fa fa-check"></i>2.1 Quality Control Analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#references"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html"><i class="fa fa-check"></i><b>7</b> biocthis introduction</a></li>
<li class="chapter" data-level="8" data-path="recount3-introduction.html"><a href="recount3-introduction.html"><i class="fa fa-check"></i><b>8</b> recount3 introduction</a></li>
<li class="chapter" data-level="9" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html"><i class="fa fa-check"></i><b>9</b> Interpreting model coefficients with ExploreModelMatrix</a></li>
<li class="chapter" data-level="10" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html"><i class="fa fa-check"></i><b>10</b> Differential gene expression analysis with limma</a></li>
<li class="chapter" data-level="11" data-path="differential-gene-expression-exercise-with-recount3-data.html"><a href="differential-gene-expression-exercise-with-recount3-data.html"><i class="fa fa-check"></i><b>11</b> Differential gene expression exercise with recount3 data</a></li>
<li class="chapter" data-level="12" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html"><i class="fa fa-check"></i><b>12</b> DE model building with variancePartition</a>
<ul>
<li class="chapter" data-level="" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html#explore-gene-level-effects"><i class="fa fa-check"></i>Explore gene-level effects</a>
<ul>
<li class="chapter" data-level="" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html#variance-partition-analysis"><i class="fa fa-check"></i>1. Variance Partition Analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html#references-1"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html"><i class="fa fa-check"></i><b>13</b> Expression heatmaps with ComplexHeatmap</a></li>
<li class="chapter" data-level="14" data-path="exercise-with-variancepartition-and-complexheatmap.html"><a href="exercise-with-variancepartition-and-complexheatmap.html"><i class="fa fa-check"></i><b>14</b> Exercise with variancePartition and ComplexHeatmap</a></li>
<li class="chapter" data-level="15" data-path="tbd.html"><a href="tbd.html"><i class="fa fa-check"></i><b>15</b> TBD</a></li>
<li class="chapter" data-level="16" data-path="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><a href="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><i class="fa fa-check"></i><b>16</b> Introduction to the world of spatially-resolved transcriptomics with Visium data</a></li>
<li class="chapter" data-level="17" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><i class="fa fa-check"></i><b>17</b> Downloading public Visium spatial data and visualizing it with spatialLIBD</a></li>
<li class="chapter" data-level="18" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><i class="fa fa-check"></i><b>18</b> Re-use of bulk RNA-seq methods for spatial data exercise</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Genome Scale Data 2023</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="summarizedexperiment-review" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> SummarizedExperiment review<a href="summarizedexperiment-review.html#summarizedexperiment-review" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Instructor: Renee</p>
<div id="overview-2" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Overview<a href="summarizedexperiment-review.html#overview-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The <code>SummarizedExperiment</code> class is used to store experimental results in the form of matrixes. Objects of this class include observations (features) of the samples, as well as additional metadata. Usually, this type of object is automatically generated as the output of other software (ie. <code>SPEAQeasy</code>), but you can also build them.</p>
<p>One of the main characteristics of <code>SummarizedExperiment</code> is that it allows you to handle you data in a “coordinated” way. For example, if you want to subset your data, with <code>SummarizedExperiment</code> you can do so without worrying your assays and metadata unsync.</p>
</div>
<div id="quiz" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Quiz<a href="summarizedexperiment-review.html#quiz" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ol style="list-style-type: decimal">
<li><p>How many classes does the <code>SummarizedExperiment</code> class has?</p></li>
<li><p>What does <strong>features</strong> stand for?</p></li>
<li><p>Which is the structure of the <code>SummarizedExperiment</code> class?</p></li>
</ol>
<figure>
<img src="Figures/se_structure.png" width="700px" align=center />
</figure>
<ol start="3" style="list-style-type: decimal">
<li><p>What type of data can we store on an assay?</p></li>
<li><p>What information does <code>colData</code> has?</p></li>
</ol>
</div>
<div id="exercises" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Exercises<a href="summarizedexperiment-review.html#exercises" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We are gonna use the same sample data set as yesterday from the <code>airway</code> library</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="summarizedexperiment-review.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SummarizedExperiment)</span>
<span id="cb8-2"><a href="summarizedexperiment-review.html#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(airway)</span>
<span id="cb8-3"><a href="summarizedexperiment-review.html#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="summarizedexperiment-review.html#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(airway, <span class="at">package =</span> <span class="st">&quot;airway&quot;</span>)</span>
<span id="cb8-5"><a href="summarizedexperiment-review.html#cb8-5" aria-hidden="true" tabindex="-1"></a>se <span class="ot">&lt;-</span> airway</span></code></pre></div>
<style>
p.exercise  {
background-color: #E4EDE2;
padding: 9px;
border: 1px solid black;
border-radius: 10px;
font-family: sans-serif;
}

</style>
<p class="exercise">
<strong>Exercise 1</strong>:
<strong>a)</strong> How many genes do we have in this object? And samples?
<strong>b)</strong> How many samples come from donors treated (<code>trt</code>) with dexamethasone (<code>dex</code>)?
</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="summarizedexperiment-review.html#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="do">## For a) you could only print the summary of the object but since the idea is to understand</span></span>
<span id="cb9-2"><a href="summarizedexperiment-review.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="do">## how to explore the object find other function that gives you the answer.</span></span>
<span id="cb9-3"><a href="summarizedexperiment-review.html#cb9-3" aria-hidden="true" tabindex="-1"></a>se</span>
<span id="cb9-4"><a href="summarizedexperiment-review.html#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; class: RangedSummarizedExperiment </span></span>
<span id="cb9-5"><a href="summarizedexperiment-review.html#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; dim: 63677 8 </span></span>
<span id="cb9-6"><a href="summarizedexperiment-review.html#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metadata(1): &#39;&#39;</span></span>
<span id="cb9-7"><a href="summarizedexperiment-review.html#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb9-8"><a href="summarizedexperiment-review.html#cb9-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rownames(63677): ENSG00000000003 ENSG00000000005 ... ENSG00000273492 ENSG00000273493</span></span>
<span id="cb9-9"><a href="summarizedexperiment-review.html#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rowData names(10): gene_id gene_name ... seq_coord_system symbol</span></span>
<span id="cb9-10"><a href="summarizedexperiment-review.html#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span></span>
<span id="cb9-11"><a href="summarizedexperiment-review.html#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colData names(9): SampleName cell ... Sample BioSample</span></span>
<span id="cb9-12"><a href="summarizedexperiment-review.html#cb9-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-13"><a href="summarizedexperiment-review.html#cb9-13" aria-hidden="true" tabindex="-1"></a><span class="do">## Same thing for b, you could just print the colData and count the samples, but this is not</span></span>
<span id="cb9-14"><a href="summarizedexperiment-review.html#cb9-14" aria-hidden="true" tabindex="-1"></a><span class="do">## efficient when our data consists in hundreds of samples. Find the answer using other tools.</span></span>
<span id="cb9-15"><a href="summarizedexperiment-review.html#cb9-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-16"><a href="summarizedexperiment-review.html#cb9-16" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(se)</span>
<span id="cb9-17"><a href="summarizedexperiment-review.html#cb9-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; DataFrame with 8 rows and 9 columns</span></span>
<span id="cb9-18"><a href="summarizedexperiment-review.html#cb9-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            SampleName     cell      dex    albut        Run avgLength Experiment    Sample    BioSample</span></span>
<span id="cb9-19"><a href="summarizedexperiment-review.html#cb9-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;   &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt;</span></span>
<span id="cb9-20"><a href="summarizedexperiment-review.html#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; SRR1039508 GSM1275862  N61311     untrt    untrt SRR1039508       126  SRX384345 SRS508568 SAMN02422669</span></span>
<span id="cb9-21"><a href="summarizedexperiment-review.html#cb9-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; SRR1039509 GSM1275863  N61311     trt      untrt SRR1039509       126  SRX384346 SRS508567 SAMN02422675</span></span>
<span id="cb9-22"><a href="summarizedexperiment-review.html#cb9-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; SRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126  SRX384349 SRS508571 SAMN02422678</span></span>
<span id="cb9-23"><a href="summarizedexperiment-review.html#cb9-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; SRR1039513 GSM1275867  N052611    trt      untrt SRR1039513        87  SRX384350 SRS508572 SAMN02422670</span></span>
<span id="cb9-24"><a href="summarizedexperiment-review.html#cb9-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 4 rows ]</span></span></code></pre></div>
<p class="exercise">
<strong>Exercise 2</strong>:
Add another assay that has the log10 of your original counts
</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="summarizedexperiment-review.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="do">## In our object, if you look at the part that says assays, we can see that at the moment</span></span>
<span id="cb10-2"><a href="summarizedexperiment-review.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="do">## we only have one with the name &quot;counts&quot;</span></span>
<span id="cb10-3"><a href="summarizedexperiment-review.html#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="summarizedexperiment-review.html#cb10-4" aria-hidden="true" tabindex="-1"></a>se</span>
<span id="cb10-5"><a href="summarizedexperiment-review.html#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; class: RangedSummarizedExperiment </span></span>
<span id="cb10-6"><a href="summarizedexperiment-review.html#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; dim: 63677 8 </span></span>
<span id="cb10-7"><a href="summarizedexperiment-review.html#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; metadata(1): &#39;&#39;</span></span>
<span id="cb10-8"><a href="summarizedexperiment-review.html#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; assays(1): counts</span></span>
<span id="cb10-9"><a href="summarizedexperiment-review.html#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rownames(63677): ENSG00000000003 ENSG00000000005 ... ENSG00000273492 ENSG00000273493</span></span>
<span id="cb10-10"><a href="summarizedexperiment-review.html#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; rowData names(10): gene_id gene_name ... seq_coord_system symbol</span></span>
<span id="cb10-11"><a href="summarizedexperiment-review.html#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span></span>
<span id="cb10-12"><a href="summarizedexperiment-review.html#cb10-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; colData names(9): SampleName cell ... Sample BioSample</span></span>
<span id="cb10-13"><a href="summarizedexperiment-review.html#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="summarizedexperiment-review.html#cb10-14" aria-hidden="true" tabindex="-1"></a><span class="do">## To see the data that&#39;s stored in that assay you can do either one of the next commands</span></span>
<span id="cb10-15"><a href="summarizedexperiment-review.html#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(se)</span>
<span id="cb10-16"><a href="summarizedexperiment-review.html#cb10-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb10-17"><a href="summarizedexperiment-review.html#cb10-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000003        679        448        873        408       1138       1047        770        572</span></span>
<span id="cb10-18"><a href="summarizedexperiment-review.html#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000005          0          0          0          0          0          0          0          0</span></span>
<span id="cb10-19"><a href="summarizedexperiment-review.html#cb10-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000419        467        515        621        365        587        799        417        508</span></span>
<span id="cb10-20"><a href="summarizedexperiment-review.html#cb10-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000457        260        211        263        164        245        331        233        229</span></span>
<span id="cb10-21"><a href="summarizedexperiment-review.html#cb10-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000460         60         55         40         35         78         63         76         60</span></span>
<span id="cb10-22"><a href="summarizedexperiment-review.html#cb10-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000938          0          0          2          0          1          0          0          0</span></span>
<span id="cb10-23"><a href="summarizedexperiment-review.html#cb10-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 63671 rows ]</span></span>
<span id="cb10-24"><a href="summarizedexperiment-review.html#cb10-24" aria-hidden="true" tabindex="-1"></a><span class="fu">assays</span>(se)<span class="sc">$</span>counts</span>
<span id="cb10-25"><a href="summarizedexperiment-review.html#cb10-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb10-26"><a href="summarizedexperiment-review.html#cb10-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000003        679        448        873        408       1138       1047        770        572</span></span>
<span id="cb10-27"><a href="summarizedexperiment-review.html#cb10-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000005          0          0          0          0          0          0          0          0</span></span>
<span id="cb10-28"><a href="summarizedexperiment-review.html#cb10-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000419        467        515        621        365        587        799        417        508</span></span>
<span id="cb10-29"><a href="summarizedexperiment-review.html#cb10-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000457        260        211        263        164        245        331        233        229</span></span>
<span id="cb10-30"><a href="summarizedexperiment-review.html#cb10-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000460         60         55         40         35         78         63         76         60</span></span>
<span id="cb10-31"><a href="summarizedexperiment-review.html#cb10-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000938          0          0          2          0          1          0          0          0</span></span>
<span id="cb10-32"><a href="summarizedexperiment-review.html#cb10-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 63671 rows ]</span></span>
<span id="cb10-33"><a href="summarizedexperiment-review.html#cb10-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-34"><a href="summarizedexperiment-review.html#cb10-34" aria-hidden="true" tabindex="-1"></a><span class="do">## Note that assay() does not support $ operator</span></span>
<span id="cb10-35"><a href="summarizedexperiment-review.html#cb10-35" aria-hidden="true" tabindex="-1"></a><span class="co"># assay(se)$counts</span></span>
<span id="cb10-36"><a href="summarizedexperiment-review.html#cb10-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-37"><a href="summarizedexperiment-review.html#cb10-37" aria-hidden="true" tabindex="-1"></a><span class="do">## We would have to do:</span></span>
<span id="cb10-38"><a href="summarizedexperiment-review.html#cb10-38" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(se, <span class="dv">1</span>)</span>
<span id="cb10-39"><a href="summarizedexperiment-review.html#cb10-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb10-40"><a href="summarizedexperiment-review.html#cb10-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000003        679        448        873        408       1138       1047        770        572</span></span>
<span id="cb10-41"><a href="summarizedexperiment-review.html#cb10-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000005          0          0          0          0          0          0          0          0</span></span>
<span id="cb10-42"><a href="summarizedexperiment-review.html#cb10-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000419        467        515        621        365        587        799        417        508</span></span>
<span id="cb10-43"><a href="summarizedexperiment-review.html#cb10-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000457        260        211        263        164        245        331        233        229</span></span>
<span id="cb10-44"><a href="summarizedexperiment-review.html#cb10-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000460         60         55         40         35         78         63         76         60</span></span>
<span id="cb10-45"><a href="summarizedexperiment-review.html#cb10-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000938          0          0          2          0          1          0          0          0</span></span>
<span id="cb10-46"><a href="summarizedexperiment-review.html#cb10-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 63671 rows ]</span></span>
<span id="cb10-47"><a href="summarizedexperiment-review.html#cb10-47" aria-hidden="true" tabindex="-1"></a><span class="fu">assay</span>(se, <span class="st">&quot;counts&quot;</span>)</span>
<span id="cb10-48"><a href="summarizedexperiment-review.html#cb10-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span></span>
<span id="cb10-49"><a href="summarizedexperiment-review.html#cb10-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000003        679        448        873        408       1138       1047        770        572</span></span>
<span id="cb10-50"><a href="summarizedexperiment-review.html#cb10-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000005          0          0          0          0          0          0          0          0</span></span>
<span id="cb10-51"><a href="summarizedexperiment-review.html#cb10-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000419        467        515        621        365        587        799        417        508</span></span>
<span id="cb10-52"><a href="summarizedexperiment-review.html#cb10-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000457        260        211        263        164        245        331        233        229</span></span>
<span id="cb10-53"><a href="summarizedexperiment-review.html#cb10-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000460         60         55         40         35         78         63         76         60</span></span>
<span id="cb10-54"><a href="summarizedexperiment-review.html#cb10-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ENSG00000000938          0          0          2          0          1          0          0          0</span></span>
<span id="cb10-55"><a href="summarizedexperiment-review.html#cb10-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 63671 rows ]</span></span>
<span id="cb10-56"><a href="summarizedexperiment-review.html#cb10-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-57"><a href="summarizedexperiment-review.html#cb10-57" aria-hidden="true" tabindex="-1"></a><span class="do">## If you use assays() without specifying the element you want to see it shows you the length</span></span>
<span id="cb10-58"><a href="summarizedexperiment-review.html#cb10-58" aria-hidden="true" tabindex="-1"></a><span class="do">## of the list and the name of each element</span></span>
<span id="cb10-59"><a href="summarizedexperiment-review.html#cb10-59" aria-hidden="true" tabindex="-1"></a><span class="fu">assays</span>(se)</span>
<span id="cb10-60"><a href="summarizedexperiment-review.html#cb10-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; List of length 1</span></span>
<span id="cb10-61"><a href="summarizedexperiment-review.html#cb10-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; names(1): counts</span></span>
<span id="cb10-62"><a href="summarizedexperiment-review.html#cb10-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-63"><a href="summarizedexperiment-review.html#cb10-63" aria-hidden="true" tabindex="-1"></a><span class="do">## To obtain a list of names as a vector you can do:</span></span>
<span id="cb10-64"><a href="summarizedexperiment-review.html#cb10-64" aria-hidden="true" tabindex="-1"></a><span class="fu">assayNames</span>(se)</span>
<span id="cb10-65"><a href="summarizedexperiment-review.html#cb10-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;counts&quot;</span></span>
<span id="cb10-66"><a href="summarizedexperiment-review.html#cb10-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-67"><a href="summarizedexperiment-review.html#cb10-67" aria-hidden="true" tabindex="-1"></a><span class="do">## Which can also be use to change the name of the assays</span></span>
<span id="cb10-68"><a href="summarizedexperiment-review.html#cb10-68" aria-hidden="true" tabindex="-1"></a><span class="fu">assayNames</span>(se)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;foo&quot;</span></span>
<span id="cb10-69"><a href="summarizedexperiment-review.html#cb10-69" aria-hidden="true" tabindex="-1"></a><span class="fu">assayNames</span>(se)</span>
<span id="cb10-70"><a href="summarizedexperiment-review.html#cb10-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;foo&quot;</span></span>
<span id="cb10-71"><a href="summarizedexperiment-review.html#cb10-71" aria-hidden="true" tabindex="-1"></a><span class="fu">assayNames</span>(se)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;counts&quot;</span></span></code></pre></div>
<p class="exercise">
<strong>Exercise 3</strong>:
Explore the metadata and add a new column that has the library size of each sample.
</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="summarizedexperiment-review.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">## To calculate the library size use</span></span>
<span id="cb11-2"><a href="summarizedexperiment-review.html#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="summarizedexperiment-review.html#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">apply</span>(<span class="fu">assay</span>(se), <span class="dv">2</span>, sum)</span>
<span id="cb11-4"><a href="summarizedexperiment-review.html#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521 </span></span>
<span id="cb11-5"><a href="summarizedexperiment-review.html#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   20637971   18809481   25348649   15163415   24448408   30818215   19126151   21164133</span></span></code></pre></div>
<pre><code>#&gt; ─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  setting  value
#&gt;  version  R version 4.3.0 (2023-04-21)
#&gt;  os       Ubuntu 22.04.2 LTS
#&gt;  system   x86_64, linux-gnu
#&gt;  ui       X11
#&gt;  language (EN)
#&gt;  collate  en_US.UTF-8
#&gt;  ctype    en_US.UTF-8
#&gt;  tz       UTC
#&gt;  date     2023-07-02
#&gt;  pandoc   2.19.2 @ /usr/local/bin/ (via rmarkdown)
#&gt; 
#&gt; ─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────
#&gt;  package                * version   date (UTC) lib source
#&gt;  airway                 * 1.20.0    2023-04-27 [1] Bioconductor
#&gt;  AnnotationDbi            1.62.1    2023-05-02 [1] Bioconductor
#&gt;  AnnotationHub            3.8.0     2023-04-25 [1] Bioconductor
#&gt;  aod                      1.3.2     2022-04-02 [1] RSPM (R 4.3.0)
#&gt;  attempt                  0.3.1     2020-05-03 [1] RSPM (R 4.3.0)
#&gt;  backports                1.4.1     2021-12-13 [1] RSPM (R 4.3.0)
#&gt;  base64enc                0.1-3     2015-07-28 [2] RSPM (R 4.3.0)
#&gt;  beachmat                 2.16.0    2023-04-25 [1] Bioconductor
#&gt;  beeswarm                 0.4.0     2021-06-01 [1] RSPM (R 4.3.0)
#&gt;  benchmarkme              1.0.8     2022-06-12 [1] RSPM (R 4.3.0)
#&gt;  benchmarkmeData          1.0.4     2020-04-23 [1] RSPM (R 4.3.0)
#&gt;  bibtex                   0.5.1     2023-01-26 [1] RSPM (R 4.3.0)
#&gt;  Biobase                * 2.60.0    2023-04-25 [1] Bioconductor
#&gt;  BiocFileCache          * 2.8.0     2023-04-25 [1] Bioconductor
#&gt;  BiocGenerics           * 0.46.0    2023-04-25 [1] Bioconductor
#&gt;  BiocIO                   1.10.0    2023-04-25 [1] Bioconductor
#&gt;  BiocManager              1.30.21   2023-06-10 [1] RSPM (R 4.3.0)
#&gt;  BiocNeighbors            1.18.0    2023-04-25 [1] Bioconductor
#&gt;  BiocParallel           * 1.34.2    2023-05-22 [1] Bioconductor
#&gt;  BiocSingular             1.16.0    2023-04-25 [1] Bioconductor
#&gt;  BiocStyle              * 2.28.0    2023-04-25 [1] Bioconductor
#&gt;  biocthis               * 1.10.3    2023-06-07 [1] Bioconductor
#&gt;  BiocVersion              3.17.1    2022-11-04 [2] Bioconductor
#&gt;  Biostrings               2.68.1    2023-05-16 [1] Bioconductor
#&gt;  bit                      4.0.5     2022-11-15 [1] RSPM (R 4.3.0)
#&gt;  bit64                    4.0.5     2020-08-30 [1] RSPM (R 4.3.0)
#&gt;  bitops                   1.0-7     2021-04-24 [1] RSPM (R 4.3.0)
#&gt;  blob                     1.2.4     2023-03-17 [1] RSPM (R 4.3.0)
#&gt;  bookdown                 0.34      2023-05-09 [1] RSPM (R 4.3.0)
#&gt;  boot                     1.3-28.1  2022-11-22 [3] CRAN (R 4.3.0)
#&gt;  broom                    1.0.5     2023-06-09 [1] RSPM (R 4.3.0)
#&gt;  bslib                    0.5.0     2023-06-09 [2] RSPM (R 4.3.0)
#&gt;  cachem                   1.0.8     2023-05-01 [2] RSPM (R 4.3.0)
#&gt;  caTools                  1.18.2    2021-03-28 [1] RSPM (R 4.3.0)
#&gt;  checkmate                2.2.0     2023-04-27 [1] RSPM (R 4.3.0)
#&gt;  circlize                 0.4.15    2022-05-10 [1] RSPM (R 4.3.0)
#&gt;  cli                      3.6.1     2023-03-23 [2] RSPM (R 4.3.0)
#&gt;  clue                     0.3-64    2023-01-31 [1] RSPM (R 4.3.0)
#&gt;  cluster                  2.1.4     2022-08-22 [3] CRAN (R 4.3.0)
#&gt;  codetools                0.2-19    2023-02-01 [3] CRAN (R 4.3.0)
#&gt;  colorspace               2.1-0     2023-01-23 [1] RSPM (R 4.3.0)
#&gt;  colourpicker             1.2.0     2022-10-28 [1] RSPM (R 4.3.0)
#&gt;  ComplexHeatmap         * 2.16.0    2023-04-25 [1] Bioconductor
#&gt;  config                   0.3.1     2020-12-17 [1] RSPM (R 4.3.0)
#&gt;  cowplot                * 1.1.1     2020-12-30 [1] RSPM (R 4.3.0)
#&gt;  crayon                   1.5.2     2022-09-29 [2] RSPM (R 4.3.0)
#&gt;  curl                     5.0.1     2023-06-07 [2] RSPM (R 4.3.0)
#&gt;  data.table               1.14.8    2023-02-17 [1] RSPM (R 4.3.0)
#&gt;  DBI                      1.1.3     2022-06-18 [1] RSPM (R 4.3.0)
#&gt;  dbplyr                 * 2.3.2     2023-03-21 [1] RSPM (R 4.3.0)
#&gt;  DelayedArray             0.26.5    2023-06-30 [1] Bioconductor
#&gt;  DelayedMatrixStats       1.22.1    2023-06-09 [1] Bioconductor
#&gt;  digest                   0.6.32    2023-06-26 [2] RSPM (R 4.3.0)
#&gt;  doParallel               1.0.17    2022-02-07 [1] RSPM (R 4.3.0)
#&gt;  dotCall64                1.0-2     2022-10-03 [1] RSPM (R 4.3.0)
#&gt;  dplyr                    1.1.2     2023-04-20 [1] RSPM (R 4.3.0)
#&gt;  dqrng                    0.3.0     2021-05-01 [1] RSPM (R 4.3.0)
#&gt;  DropletUtils             1.20.0    2023-04-25 [1] Bioconductor
#&gt;  DT                       0.28      2023-05-18 [1] RSPM (R 4.3.0)
#&gt;  edgeR                  * 3.42.4    2023-05-31 [1] Bioconductor
#&gt;  ellipsis                 0.3.2     2021-04-29 [2] RSPM (R 4.3.0)
#&gt;  EnvStats                 2.7.0     2022-03-07 [1] RSPM (R 4.3.0)
#&gt;  evaluate                 0.21      2023-05-05 [2] RSPM (R 4.3.0)
#&gt;  ExperimentHub            2.8.0     2023-04-25 [1] Bioconductor
#&gt;  ExploreModelMatrix     * 1.12.0    2023-04-25 [1] Bioconductor
#&gt;  fansi                    1.0.4     2023-01-22 [2] RSPM (R 4.3.0)
#&gt;  fastmap                  1.1.1     2023-02-24 [2] RSPM (R 4.3.0)
#&gt;  fields                   14.1      2022-08-12 [1] RSPM (R 4.3.0)
#&gt;  filelock                 1.0.2     2018-10-05 [1] RSPM (R 4.3.0)
#&gt;  foreach                  1.5.2     2022-02-02 [1] RSPM (R 4.3.0)
#&gt;  foreign                  0.8-84    2022-12-06 [3] CRAN (R 4.3.0)
#&gt;  Formula                  1.2-5     2023-02-24 [1] RSPM (R 4.3.0)
#&gt;  fs                       1.6.2     2023-04-25 [2] RSPM (R 4.3.0)
#&gt;  generics                 0.1.3     2022-07-05 [1] RSPM (R 4.3.0)
#&gt;  GenomeInfoDb           * 1.36.1    2023-06-21 [1] Bioconductor
#&gt;  GenomeInfoDbData         1.2.10    2023-05-17 [1] Bioconductor
#&gt;  GenomicAlignments        1.36.0    2023-04-25 [1] Bioconductor
#&gt;  GenomicRanges          * 1.52.0    2023-04-25 [1] Bioconductor
#&gt;  GetoptLong               1.0.5     2020-12-15 [1] RSPM (R 4.3.0)
#&gt;  ggbeeswarm               0.7.2     2023-04-29 [1] RSPM (R 4.3.0)
#&gt;  ggplot2                * 3.4.2     2023-04-03 [1] RSPM (R 4.3.0)
#&gt;  ggrepel                * 0.9.3     2023-02-03 [1] RSPM (R 4.3.0)
#&gt;  GlobalOptions            0.1.2     2020-06-10 [1] RSPM (R 4.3.0)
#&gt;  glue                     1.6.2     2022-02-24 [2] RSPM (R 4.3.0)
#&gt;  golem                    0.4.1     2023-06-05 [1] RSPM (R 4.3.0)
#&gt;  gplots                   3.1.3     2022-04-25 [1] RSPM (R 4.3.0)
#&gt;  gridExtra                2.3       2017-09-09 [1] RSPM (R 4.3.0)
#&gt;  gtable                   0.3.3     2023-03-21 [1] RSPM (R 4.3.0)
#&gt;  gtools                   3.9.4     2022-11-27 [1] RSPM (R 4.3.0)
#&gt;  HDF5Array                1.28.1    2023-05-01 [1] Bioconductor
#&gt;  here                   * 1.0.1     2020-12-13 [1] RSPM (R 4.3.0)
#&gt;  Hmisc                  * 5.1-0     2023-05-08 [1] RSPM (R 4.3.0)
#&gt;  hms                      1.1.3     2023-03-21 [1] RSPM (R 4.3.0)
#&gt;  htmlTable                2.4.1     2022-07-07 [1] RSPM (R 4.3.0)
#&gt;  htmltools                0.5.5     2023-03-23 [2] RSPM (R 4.3.0)
#&gt;  htmlwidgets              1.6.2     2023-03-17 [2] RSPM (R 4.3.0)
#&gt;  httpuv                   1.6.11    2023-05-11 [2] RSPM (R 4.3.0)
#&gt;  httr                     1.4.6     2023-05-08 [2] RSPM (R 4.3.0)
#&gt;  igraph                   1.5.0     2023-06-16 [1] RSPM (R 4.3.0)
#&gt;  interactiveDisplayBase   1.38.0    2023-04-25 [1] Bioconductor
#&gt;  IRanges                * 2.34.1    2023-06-22 [1] Bioconductor
#&gt;  irlba                    2.3.5.1   2022-10-03 [1] RSPM (R 4.3.0)
#&gt;  iSEE                   * 2.12.0    2023-04-25 [1] Bioconductor
#&gt;  iterators                1.0.14    2022-02-05 [1] RSPM (R 4.3.0)
#&gt;  jquerylib                0.1.4     2021-04-26 [2] RSPM (R 4.3.0)
#&gt;  jsonlite                 1.8.7     2023-06-29 [2] RSPM (R 4.3.0)
#&gt;  KEGGREST                 1.40.0    2023-04-25 [1] Bioconductor
#&gt;  KernSmooth               2.23-21   2023-05-03 [3] RSPM (R 4.3.0)
#&gt;  knitr                    1.43      2023-05-25 [2] RSPM (R 4.3.0)
#&gt;  later                    1.3.1     2023-05-02 [2] RSPM (R 4.3.0)
#&gt;  lattice                  0.21-8    2023-04-05 [3] CRAN (R 4.3.0)
#&gt;  lazyeval                 0.2.2     2019-03-15 [1] RSPM (R 4.3.0)
#&gt;  lifecycle                1.0.3     2022-10-07 [2] RSPM (R 4.3.0)
#&gt;  limma                  * 3.56.2    2023-06-04 [1] Bioconductor
#&gt;  lme4                     1.1-33    2023-04-25 [1] RSPM (R 4.3.0)
#&gt;  lobstr                 * 1.1.2     2022-06-22 [1] RSPM (R 4.3.0)
#&gt;  locfit                   1.5-9.8   2023-06-11 [1] RSPM (R 4.3.0)
#&gt;  lubridate                1.9.2     2023-02-10 [1] RSPM (R 4.3.0)
#&gt;  magick                   2.7.4     2023-03-09 [1] RSPM (R 4.3.0)
#&gt;  magrittr                 2.0.3     2022-03-30 [2] RSPM (R 4.3.0)
#&gt;  maps                     3.4.1     2022-10-30 [1] RSPM (R 4.3.0)
#&gt;  MASS                     7.3-60    2023-05-04 [3] RSPM (R 4.3.0)
#&gt;  Matrix                   1.5-4.1   2023-05-18 [3] RSPM (R 4.3.0)
#&gt;  MatrixGenerics         * 1.12.2    2023-06-09 [1] Bioconductor
#&gt;  matrixStats            * 1.0.0     2023-06-02 [1] RSPM (R 4.3.0)
#&gt;  memoise                  2.0.1     2021-11-26 [2] RSPM (R 4.3.0)
#&gt;  mgcv                     1.8-42    2023-03-02 [3] CRAN (R 4.3.0)
#&gt;  mime                     0.12      2021-09-28 [2] RSPM (R 4.3.0)
#&gt;  miniUI                   0.1.1.1   2018-05-18 [2] RSPM (R 4.3.0)
#&gt;  minqa                    1.2.5     2022-10-19 [1] RSPM (R 4.3.0)
#&gt;  munsell                  0.5.0     2018-06-12 [1] RSPM (R 4.3.0)
#&gt;  mvtnorm                  1.2-2     2023-06-08 [1] RSPM (R 4.3.0)
#&gt;  nlme                     3.1-162   2023-01-31 [3] CRAN (R 4.3.0)
#&gt;  nloptr                   2.0.3     2022-05-26 [1] RSPM (R 4.3.0)
#&gt;  nnet                     7.3-19    2023-05-03 [3] RSPM (R 4.3.0)
#&gt;  paletteer                1.5.0     2022-10-19 [1] RSPM (R 4.3.0)
#&gt;  patchwork              * 1.1.2     2022-08-19 [1] RSPM (R 4.3.0)
#&gt;  pbkrtest                 0.5.2     2023-01-19 [1] RSPM (R 4.3.0)
#&gt;  pheatmap               * 1.0.12    2019-01-04 [1] RSPM (R 4.3.0)
#&gt;  pillar                   1.9.0     2023-03-22 [2] RSPM (R 4.3.0)
#&gt;  pkgconfig                2.0.3     2019-09-22 [2] RSPM (R 4.3.0)
#&gt;  plotly                   4.10.2    2023-06-03 [1] RSPM (R 4.3.0)
#&gt;  plyr                     1.8.8     2022-11-11 [1] RSPM (R 4.3.0)
#&gt;  png                      0.1-8     2022-11-29 [1] RSPM (R 4.3.0)
#&gt;  postcards              * 0.2.3     2022-01-07 [1] RSPM (R 4.3.0)
#&gt;  prettyunits              1.1.1     2020-01-24 [2] RSPM (R 4.3.0)
#&gt;  progress                 1.2.2     2019-05-16 [1] RSPM (R 4.3.0)
#&gt;  promises                 1.2.0.1   2021-02-11 [2] RSPM (R 4.3.0)
#&gt;  purrr                    1.0.1     2023-01-10 [2] RSPM (R 4.3.0)
#&gt;  R.cache                  0.16.0    2022-07-21 [1] RSPM (R 4.3.0)
#&gt;  R.methodsS3              1.8.2     2022-06-13 [1] RSPM (R 4.3.0)
#&gt;  R.oo                     1.25.0    2022-06-12 [1] RSPM (R 4.3.0)
#&gt;  R.utils                  2.12.2    2022-11-11 [1] RSPM (R 4.3.0)
#&gt;  R6                       2.5.1     2021-08-19 [2] RSPM (R 4.3.0)
#&gt;  rappdirs                 0.3.3     2021-01-31 [2] RSPM (R 4.3.0)
#&gt;  rbibutils                2.2.13    2023-01-13 [1] RSPM (R 4.3.0)
#&gt;  RColorBrewer           * 1.1-3     2022-04-03 [1] RSPM (R 4.3.0)
#&gt;  Rcpp                     1.0.10    2023-01-22 [2] RSPM (R 4.3.0)
#&gt;  RCurl                    1.98-1.12 2023-03-27 [1] RSPM (R 4.3.0)
#&gt;  Rdpack                   2.4       2022-07-20 [1] RSPM (R 4.3.0)
#&gt;  recount3               * 1.10.2    2023-05-07 [1] Bioconductor
#&gt;  RefManageR             * 1.4.0     2022-09-30 [1] RSPM (R 4.3.0)
#&gt;  remaCor                  0.0.16    2023-06-21 [1] RSPM (R 4.3.0)
#&gt;  rematch2                 2.1.2     2020-05-01 [2] RSPM (R 4.3.0)
#&gt;  reshape2                 1.4.4     2020-04-09 [1] RSPM (R 4.3.0)
#&gt;  restfulr                 0.0.15    2022-06-16 [1] RSPM (R 4.3.0)
#&gt;  rhdf5                    2.44.0    2023-04-25 [1] Bioconductor
#&gt;  rhdf5filters             1.12.1    2023-04-30 [1] Bioconductor
#&gt;  Rhdf5lib                 1.22.0    2023-04-25 [1] Bioconductor
#&gt;  RhpcBLASctl              0.23-42   2023-02-11 [1] RSPM (R 4.3.0)
#&gt;  rintrojs                 0.3.2     2022-08-09 [1] RSPM (R 4.3.0)
#&gt;  rjson                    0.2.21    2022-01-09 [1] RSPM (R 4.3.0)
#&gt;  rlang                  * 1.1.1     2023-04-28 [2] RSPM (R 4.3.0)
#&gt;  rmarkdown                2.22      2023-06-01 [2] RSPM (R 4.3.0)
#&gt;  rpart                    4.1.19    2022-10-21 [3] CRAN (R 4.3.0)
#&gt;  rprojroot                2.0.3     2022-04-02 [2] RSPM (R 4.3.0)
#&gt;  Rsamtools                2.16.0    2023-04-25 [1] Bioconductor
#&gt;  RSQLite                  2.3.1     2023-04-03 [1] RSPM (R 4.3.0)
#&gt;  rstudioapi               0.14      2022-08-22 [2] RSPM (R 4.3.0)
#&gt;  rsvd                     1.0.5     2021-04-16 [1] RSPM (R 4.3.0)
#&gt;  rtracklayer              1.60.0    2023-04-25 [1] Bioconductor
#&gt;  S4Arrays                 1.0.4     2023-05-14 [1] Bioconductor
#&gt;  S4Vectors              * 0.38.1    2023-05-02 [1] Bioconductor
#&gt;  sass                     0.4.6     2023-05-03 [2] RSPM (R 4.3.0)
#&gt;  ScaledMatrix             1.8.1     2023-05-03 [1] Bioconductor
#&gt;  scales                   1.2.1     2022-08-20 [1] RSPM (R 4.3.0)
#&gt;  scater                 * 1.28.0    2023-04-25 [1] Bioconductor
#&gt;  scuttle                * 1.10.1    2023-05-02 [1] Bioconductor
#&gt;  sessioninfo            * 1.2.2     2021-12-06 [2] RSPM (R 4.3.0)
#&gt;  shape                    1.4.6     2021-05-19 [1] RSPM (R 4.3.0)
#&gt;  shiny                    1.7.4     2022-12-15 [2] RSPM (R 4.3.0)
#&gt;  shinyAce                 0.4.2     2022-05-06 [1] RSPM (R 4.3.0)
#&gt;  shinydashboard           0.7.2     2021-09-30 [1] RSPM (R 4.3.0)
#&gt;  shinyjs                  2.1.0     2021-12-23 [1] RSPM (R 4.3.0)
#&gt;  shinyWidgets             0.7.6     2023-01-08 [1] RSPM (R 4.3.0)
#&gt;  SingleCellExperiment   * 1.22.0    2023-04-25 [1] Bioconductor
#&gt;  smokingMouse           * 0.99.5    2023-06-28 [1] Github (LieberInstitute/smokingMouse@2e7640c)
#&gt;  spam                     2.9-1     2022-08-07 [1] RSPM (R 4.3.0)
#&gt;  sparseMatrixStats        1.12.1    2023-06-20 [1] Bioconductor
#&gt;  SpatialExperiment      * 1.10.0    2023-04-25 [1] Bioconductor
#&gt;  spatialLIBD            * 1.12.0    2023-04-27 [1] Bioconductor
#&gt;  statmod                  1.5.0     2023-01-06 [1] RSPM (R 4.3.0)
#&gt;  stringi                  1.7.12    2023-01-11 [2] RSPM (R 4.3.0)
#&gt;  stringr                * 1.5.0     2022-12-02 [2] RSPM (R 4.3.0)
#&gt;  styler                   1.10.1    2023-06-05 [1] RSPM (R 4.3.0)
#&gt;  SummarizedExperiment   * 1.30.2    2023-06-06 [1] Bioconductor
#&gt;  tibble                   3.2.1     2023-03-20 [2] RSPM (R 4.3.0)
#&gt;  tidyr                    1.3.0     2023-01-24 [1] RSPM (R 4.3.0)
#&gt;  tidyselect               1.2.0     2022-10-10 [1] RSPM (R 4.3.0)
#&gt;  timechange               0.2.0     2023-01-11 [1] RSPM (R 4.3.0)
#&gt;  usethis                * 2.2.1     2023-06-23 [2] RSPM (R 4.3.0)
#&gt;  utf8                     1.2.3     2023-01-31 [2] RSPM (R 4.3.0)
#&gt;  variancePartition      * 1.30.2    2023-06-07 [1] Bioconductor
#&gt;  vctrs                    0.6.3     2023-06-14 [2] RSPM (R 4.3.0)
#&gt;  vipor                    0.4.5     2017-03-22 [1] RSPM (R 4.3.0)
#&gt;  viridis                  0.6.3     2023-05-03 [1] RSPM (R 4.3.0)
#&gt;  viridisLite              0.4.2     2023-05-02 [1] RSPM (R 4.3.0)
#&gt;  withr                    2.5.0     2022-03-03 [2] RSPM (R 4.3.0)
#&gt;  xfun                     0.39      2023-04-20 [2] RSPM (R 4.3.0)
#&gt;  XML                      3.99-0.14 2023-03-19 [1] RSPM (R 4.3.0)
#&gt;  xml2                     1.3.4     2023-04-27 [2] RSPM (R 4.3.0)
#&gt;  xtable                   1.8-4     2019-04-21 [2] RSPM (R 4.3.0)
#&gt;  XVector                  0.40.0    2023-04-25 [1] Bioconductor
#&gt;  yaml                     2.3.7     2023-01-23 [2] RSPM (R 4.3.0)
#&gt;  zlibbioc                 1.46.0    2023-04-25 [1] Bioconductor
#&gt; 
#&gt;  [1] /__w/_temp/Library
#&gt;  [2] /usr/local/lib/R/site-library
#&gt;  [3] /usr/local/lib/R/library
#&gt; 
#&gt; ──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</code></pre>

</div>
</div>
<div class="container">
    <footer class="site-footer">

      <center>
<p class="powered-by">
© 2011-2020. All thoughts and opinions here are my own. The icon was designed by <a href="https://www.linkedin.com/in/mauricio-guzman-6529b551/">Mauricio Guzmán</a> and is inspired by <a href="https://en.wikipedia.org/wiki/Huichol">Huichol culture</a>; it represents my community building interests.
</p>

<p class="powered-by copyright-license-text">
This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">CC BY NC SA 4.0</a>
</p>

<p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">
      <img src="https://search.creativecommons.org/static/img/cc_icon.svg" alt="CC icon" width="32px">
      <img src="https://search.creativecommons.org/static/img/cc-by_icon.svg" alt="CC by icon" width="32px">
      <img src="https://search.creativecommons.org/static/img/cc-nc_icon.svg" alt="CC NC icon" width="32px">
      <img src="https://search.creativecommons.org/static/img/cc-sa_icon.svg" alt="CC SA icon" width="32px">
    </a>
</p>

<div id="clustrmaps-widget"></div><script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=300&t=m&d=Zo-deincaJGsqhqjIWsujtdoNOMJsnnZLefKmxDKVv8&co=124b73&cmo=dd53ff&cmn=4ec724'></script>

<p class="powered-by">
Published with <a href="https://bookdown.org/" target="_blank" rel="noopener">Bookdown</a>
</p>

</center>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-45299226-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-45299226-1');
</script>

</footer>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="smokingmouse-study.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="smokingmouse-rse.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
